README.md

```diff
@@ -1,2 +1,437 @@
-# python-query-filter
-Python's `filter` function with composable queries. Filter collections of objects and nested dictionaries.
+## python-query-filter
+
+This package provides a functional API for filtering collections of
+heterogeneous, nested dictionaries or complex objects. It has 100% test
+coverage.
+
+At the core of this API are `q_filter` functions, which are like
+the built-in `filter` function, but take any number of predicate functions
+rather than just one.
+
+The remainder of the functions in this package
+are used to construct predicates that evaluate items
+or attributes within filtered objects.
+
+### Use Case
+This package is best suited to nested, heterogeneous data. Collections
+of flat homogenous dictionaries or objects are easier to filter with
```

Comment:
It is not uncommon in the context of computing to write any of "heterogeneous", "heterogenous", "homogenous", and "homogeneous" when referring to data structures, so it's not clear that any of these can be called incorrect. However, I do think that it would make sense to be consistent with the "-geneous" or "-genous" suffix.

Reasons to prefer "-geneous":
- "-genous" is proscribed or discouraged by some dictionaries, such as the Oxford English Dictionary.
  - But not by the Merriam-Webster dictionary, for example.
  - But this only means it it proscribed in general usage; many technical terms do not follow general spelling rules.
  - But going against the general spelling rule makes the whole language less consistent and confuses people.
- There are about 14,600 and 3,630 Google search results for "homogenous list" and "heterogenous list", but 107,000 and 1,230,000 for "homogeneous list" and "heterogeneous list", respectively, indicating that the latter are more widely accepted even in this particular context.

Reasons to prefer "-genous":
- It is shorter, saving space and time.
- It is spelt more phonetically with respect to how the words are typically pronounced in this context.

After considering the above points, I would personally prefer the former choice.

Reply:
Thanks for the detailed argument. I'm inclined to agree with you.